Update metrics-server deployment apiVersion to apps/v1

elisiano opened the pull request:
Since kubernetes 1.16 does not support older deployments api versions
this needs to be updated as well.

elisiano commented:
I realize there are old installations in the wild, would it be better to have a separate manifest called `v1.16.x.yaml` instead?

johngmyers commented:
Yes, there should be a separate manifest.

elisiano commented:
@johngmyers done, let me know if I should address anything ellse.

johngmyers commented:
I think you might need to add an `addon.yaml` to the directory, specifying the `kubernetesVersion` constraints.

johngmyers commented:
/ok-to-test

elisiano commented:
@johngmyers I've added the `addon.yaml` and fixed the `README.md` to reflect the cluster version constraints.

Please note that I'm not a user of addons, so I would need someone to verify that the file format is correct.

justinsb commented:
Thanks @elisiano - this looks great!

/approve
/lgtm

johngmyers commented:
/retest
